buildenv/jenkins/jobs/infrastructure/copyrightCheck

```diff
@@ -36,386 +39,405 @@ def ignorefile = "${env.WORKSPACE}/.copyrightignore"
-                        REGEX = "\'Copyright \\(c\\) ([0-9]{4}), ${DATE_YEAR}\'"
-                    } else if (ghprbGhRepository ==~ "ibmruntimes.*") {
-                        REGEX = "\'\\(c\\) Copyright IBM Corp. ([0-9]{4}), ${DATE_YEAR} All Rights Reserved\'"
+                try {
+                    def remoteConfigParameters = [refspec: "+refs/pull/${ghprbPullId}/*:refs/remotes/origin/pr/${ghprbPullId}/* +refs/heads/${ghprbTargetBranch}:refs/remotes/origin/${ghprbTargetBranch}",
+                                                      url: SRC_REPO]
+                    if (GIT_CREDENTIALS_ID) {
+                        remoteConfigParameters.put("credentialsId", "${GIT_CREDENTIALS_ID}")
+                    }
+                    String REPO_CACHE_DIR = params.REPO_CACHE_DIR ?: "${env.HOME}/openjdk_cache"
+
+                    checkout changelog: false, poll: false,
+                            scm: [$class: 'GitSCM',
+                                branches: [[name: sha1]],
+                                doGenerateSubmoduleConfigurations: false,
+                                extensions: [[$class: 'CloneOption',
+                                                depth: 0,
+                                                noTags: false,
+                                                reference: "${REPO_CACHE_DIR}",
+                                                shallow: false]],
+                                userRemoteConfigs: [remoteConfigParameters]]
+
+                    if (GIT_CREDENTIALS_ID) {
+                        sshagent(credentials:["${GIT_CREDENTIALS_ID}"]) {
+                            FILES = get_files()
+                        }
                     } else {
-                        error("Unrecognized repository. Unable to determine correct Copyright regex")
+                        FILES = get_files()
                     }
+                    echo FILES
+                    if (FILES == "") {
+                        echo "There are no files to check for copyrights"
+                    } else {
+                        def FILES_LIST = FILES.split("\\r?\\n")
+                        DATE_YEAR = sh (
+                            script: "date +%Y",
+                            returnStdout: true
+                        ).trim()
 
-                    if (fileExists("${env.WORKSPACE}/.copyrightignore")) {
-                        def readinFiles = readFile("${env.WORKSPACE}/.copyrightignore")
-                        def READLIST = readinFiles.split("\\r?\\n")
-                        for (item in READLIST) {
-                            regexsElm = item
+                        // Set a different Copyright regex depending on the Repo the PR is from
+                        if (ghprbGhRepository ==~ "eclipse.*") {
+                            REGEX = "\'Copyright \\(c\\) ([0-9]{4}), ${DATE_YEAR}\'"
+                        } else if (ghprbGhRepository.tokenize('/').get(0).endsWith('runtimes')) {
```

Comment:
I think this block of code "Set a different Copyright regex" is a carry-over from when we did all the copyrights out of the OpenJ9 file. We can probably strip out all this and just leave the REGEX set to the IBM one.

Reply:
Cleaned up.